modules/core/src/lifecycle/prop-types.ts

```diff
@@ -57,6 +57,8 @@ type ImagePropType = BasePropType & {
 };
 type ObjectPropType = BasePropType & {
   type: 'object';
+  optional?: boolean;
+  compare?: boolean;
```

Comment:
Since we are at this, `loadOptions ` could also cause an update by shallow comparison. We could add an option to the prop type that directs diffing to ignore shallow updates in loadOptions because it's not used until some other async prop changes.

Reply:
I would do this in a followup PR